### PR TITLE
Bugfix TypeError

### PR DESCRIPTION
**Pull request by krocheck**
Cannot read property 'replace' of null

### CODE REVIEW

**Comment by krocheck** [comment]
@McHauge an exception for this was popping up in the logs.  Can you please validate this patch doesn't break anything.

**Comment by McHauge** [comment]
not too sure, don't have anything to test it with, maybe @hjoelr has something to test it on.

**Comment by hjoelr** [comment]
@krocheck @McHauge I do have a PTZ Optics camera. I can try to debug it, but may not be until this weekend. Is there any more information on when or why this was happening?

**Comment by krocheck** [comment]
I don't know the case the caused it, best guess is that people weren't filling in the custom input field yet still trying to fire the action.  Regardless, the test is to make sure custom actions still file.  I put a check inline to ignore the action if the input is undefined.

File: index.js [hunk 1]
```diff
@@ -1111,14 +1111,16 @@ instance.prototype.action = function(action) {
...
 		case 'custom':
-			var hexData = opt.custom.replace(/\s+/g, '');
-			var tempBuffer = new Buffer(hexData, 'hex');
-			cmd = tempBuffer.toString('binary');
-
-			if ((tempBuffer[0] & 0xF0) === 0x80) {
-				self.sendVISCACommand(cmd);
-			} else {
-				self.log('error', 'Error, command "' + opt.custom + '" does not start with 8');
+			if (typeof opt.custom === 'string' || opt.custom instanceof String) {
```
Comment (hjoelr) [inline index.js:1114]: @krocheck What about logging an error when the opt.custom is not a string instead of just silently ignoring it?

**Review by hjoelr — CHANGES_REQUESTED** [review]

**Comment by hjoelr** [comment]
@krocheck I ran your changes through and it appeared to be working fine. Here's a screenshot in the middle of the debugging.

image

**Comment by hjoelr** [comment]
@krocheck I think I figured out why some people are getting the `Cannot read property 'replace' of null` error. I recently posted a Regex validation change (#6) for the custom command to be in line with PTZ Optics' command document. There must have been people who had commands in there that were based on the old Regex. For a test, I put a valid Regex into a button and it worked fine. I then terminated the program and intentionally changed the Regex to be something that the value did not match against and restarted the program. When I opened that button settings the field was empty. When I "ran" the button the `opt.custom` was null.

I don't think the solution would be to revert the Regex change. I'm not sure how it would have even worked before as the previous Regex no way matched the command syntax provided by PTZ Optics. I think logging the error might be the best thing to alert the people of the issue with their commands.

**Comment by hjoelr** [comment]
I guess another option would be to update the current regex to support both the old regex or the new (correct) one. I wish there was a way to know if the custom command even worked before as the old regex would not properly match the commands the camera supports.

**Comment by krocheck** [comment]
Does PTZ Optics not permit/read camera IDs >1 which is why you change the first byte to 81 only?

**Comment by hjoelr** [comment]
@krocheck Based on [this document](https://ptzoptics.com/wp-content/uploads/2019/10/PTZOptics-VISCA-over-IP-Commands-Rev1_1.1-10-19.pdf) from PTZ Optics it always starts with `81`.

If this was visca over RS-485, the second digit would differentiate between different cameras. However, when using IP, the IP address by definition already defines which camera we're talking to. So, it wouldn't make sense to also have to reference camera number in the actual visca over IP command.

**Pull request merged**